### PR TITLE
feat(options): support htmlWebpackPlugin options & replace [name] in …

### PR DESCRIPTION
**Pull request by xiaoiver**
1. support other htmlWebpackPlugin options, like inject, favicon
2. also replace [name] in htmlTemplatePath

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@xiaoiver This is a really great idea to have an escape hatch for the rest of the options.

Could you please add a few test cases for the specific features you are leveraging from htmlWebpackPluginOptions?

File: src/plugin.js [hunk 1]
```diff
@@ -22,7 +23,8 @@ function setPluginOptions (pluginOptions) {
     bootstrapFilename: bootstrapFilename || 'inline.chunk.js',
     templateFilename: templateFilename || 'index.html',
     templatePath: templatePath || 'templates/[name]',
-    htmlTemplatePath: htmlTemplatePath || undefined
+    htmlTemplatePath: htmlTemplatePath || undefined,
+    htmlWebpackPluginOptions: htmlWebpackPluginOptions || {}
```
Comment (TheLarkInn) [inline src/plugin.js]: I'd like to have some tests specific for a few of the features used here. I know there are a lot so just a few would be good. 

Also could I convince you to submit an issue (as a reminder to us) to add better tests for html-webpack-plugin options overall.

**Review by TheLarkInn — CHANGES_REQUESTED** [review]

File: src/plugin.js (hunk 1 above)
Comment (xiaoiver) [inline src/plugin.js]: OK, I'll add a test case for some html-webpack-plugin options maybe like favicon, minify...

**Comment by xiaoiver** [comment]
I added a simple test case for `minify` and `favicon` options in html-webpack-plugin. Is there anything I should complement？

File: src/plugin.js (hunk 1 above)
Comment (TheLarkInn) [inline src/plugin.js]: This looks wonderful. Very well done.

**Review by TheLarkInn — APPROVED** [review]

**Comment by TheLarkInn** [comment]
Thank you @xiaoiver for the contribution. We use semantic release to automatically publish so please keep an eye on master branch Travis build to see when update is published

**Pull request merged**